Automattic/jetpack
# Search: Fix search plan detection

#### Changes proposed in this Pull Request:
* Check for Search product eligibility by checking the site's purchases instead of its plan.
* Removes Instant Search feature gate for Jetpacks' Plan page. 
* **Known issue**: Enabling the Jetpack Search module without a Professional Plan is currently not possible.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No.

#### Testing instructions:
* Apply this change to your Jetpack installation.
* Using a site without a Jetpack Search purchase, navigate to the following pages below. Ensure that each page works as expected (e.g. see upsells for Jetpack Search as necessary).
  1) The dashboard (/wp-admin/admin.php?page=jetpack#/dashboard)
  2) My Plan page (/wp-admin/admin.php?page=jetpack#/my-plan)
  3) Plans page (/wp-admin/admin.php?page=jetpack#/plans)
  4) Performance settings page (/wp-admin/admin.php?page=jetpack#/performance).
* Purchase a Jetpack Search product to your site and try the four pages listed above again. Ensure that each page works as expected.

#### Proposed changelog entry for your changes:
* None
